File: challenges/multi_bracket_validation/multi_bracket_validation.py


def multi_bracket_validation(input_string):
    """
    """
    open_bracket = ["[", "{", "("]
    close_bracket = ["]", "}", ")"]
    bracket = Stack()

    for char in input_string:
        if char in open_bracket:
            bracket.push(char)

        if char in close_bracket:
            if char == ')':
                if bracket.peek() == '(':
                    bracket.pop
            elif char == '}':
                if bracket.peek() == '{':
                    bracket.pop
            elif char == ']':
                if bracket.peek() == '[':
                    bracket.pop
    if bracket.peek() is None:
        return True
    else:
        return False

File: 401d56/code-challenges/gradle/src/main/java/stacksandqueues/Multibracketvalidation.java
package stacksandqueues;

import java.util.Stack;

public class Multibracketvalidation {
    public static boolean multibracketvalidation(String input){
        String[] inputArray = input.split("");
        Stack<String> newStack = new Stack<>();
        for(int i=0; i < inputArray.length; i++){
            if(inputArray[i].equals("[") || inputArray[i].equals("{") || inputArray[i].equals("(")){
                newStack.push(inputArray[i]);
            }
            else{
                if(inputArray[i].equals("}") && newStack.peek().equals("{")){
                        newStack.pop();
                }
                else if(inputArray[i].equals("]") && newStack.peek().equals("[")){
                    newStack.pop();
                }
                else if(inputArray[i].equals(")") && newStack.peek().equals("(")){
                    newStack.pop();
                }
            }
        }
        if(newStack.isEmpty()){
            return true;
        }
        else{
            return false;
        }
    }
}